[Data] [Docs] Add Overview page for Ray Data (#36105)
ray-project/ray: doc/source/data/data.rst

Ray Data: Scalable Datasets for ML
==================================

> Ray Data is a scalable data processing library for ML workloads. It provides flexible and performant APIs for scaling :ref:`Offline batch inference <batch_inference_overview>` and :ref:`Data preprocessing and ingest for ML training <ml_ingest_overview>`

Reviewer: ```suggestion
Ray Data is a scalable data processing library for ML workloads. It provides flexible and performant APIs for scaling :ref:`Offline batch inference <batch_inference_overview>` and :ref:`Data preprocessing and ingest for ML training <ml_ingest_overview>`.
```

    .. grid-item-card::

        **User Guides**
        ^^^

        Learn how to use Ray Data, from basic usage to end-to-end guides.

        +++
        .. button-ref:: data_user_guide
            :color: primary
            :outline:
            :expand:

            Learn how to use Ray Data

    .. grid-item-card::

        **Key Concepts**
        ^^^

        Understand the key concepts behind Ray Data.

>         Learn what :ref:`Datasets <dataset_concept>` are and how they are executed in Ray Data.

Reviewer 2: ```suggestion
        Learn what :ref:`Datasets <dataset_concept>` are and how Ray Data executes them.
```

    .. grid-item-card::

        **Ray blogs**
        ^^^

        Get the latest on engineering updates from the Ray team and how companies are using Ray Data.

        +++
        .. button-link:: https://www.anyscale.com/blog?tag=ray-datasets
            :color: primary
            :outline:
            :expand:

            Read the Ray blogs

----------

> Contribute

Reviewer: Should we remove this section?

    .. grid-item-card::

        **Ray blogs**
        ^^^

>         Get the latest on engineering updates from the Ray team and how companies are using Ray Data.

Reviewer 2: ```suggestion
        Get the latest engineering updates from the Ray team and how companies are using Ray Data.
```
